### PR TITLE
Implement calculus operations for ConstantSpace

### PR DESCRIPTION
**Pull request by LucasAschenbach**
I have implemented the `integrate` and `differentiate` methods for `ConstantSpace`. This addition was motivated to add support for taking the norm of ConstantSpace functionals. I have also added a test for the norm operation.

### CODE REVIEW

File: src/Spaces/ConstantSpace.jl [hunk 1]
```diff
@@ -254,7 +254,7 @@ choosedomainspace(M::ConcreteMultiplication{D,UnsetSpace},sp::Space) where {D<:C
 Base.isfinite(f::Fun{CS}) where {CS<:ConstantSpace} = isfinite(Number(f))
 
 
+## Calculus
 
-
-
-
+integrate(f::Fun{<:ConstantSpace{<:IntervalOrSegment}}) = Number(f) * Fun(Chebyshev(domain(f)))
```
Comment (jishnub) [inline src/Spaces/ConstantSpace.jl:259]: `Chebyshev` isn't defined in this package, so perhaps you want `Fun(domain(f))`

Comment (LucasAschenbach) [reply src/Spaces/ConstantSpace.jl]: Oh yes, thanks! I've update the code.

**Comment by jishnub** [comment]
I think the test for the integral needs to be added to `ApproxFunOrthogonalPolynomials`, as that requires `Chebyshev` to be defined

**Comment by LucasAschenbach** [comment]
The problem is that the canonical domain from DomainSets is chosen as `ChebyshevInterval` but the constructor `Space(::ChebyshevInterval)` is not implemented in ApproxFunBase. This would be fixed by importing ApproxFunOrthogonalPolynomials into the test environment. Is this dependency acceptable?

**Comment by LucasAschenbach** [comment]
Got it, thanks for the review! Should I create a PR for ApproxFunOrthogonalPolynomials?

**Comment by jishnub** [comment]
Sure, but that test PR would need this to be merged first. Maybe you can bump the patch version of this package to 0.9.15 in this PR, and set the minimum compat bound for `ApproxFunBase` to 0.9.15 in the PR to `ApproxFunOrthogonalPolynomials`

**Pull request merged**